frost-core/src/frost/identifier.rs

```diff
@@ -14,183 +14,87 @@ use crate::{Ciphersuite, Error, Field, Group, Scalar};
 /// over, corresponding to some x-coordinate for a polynomial f(x) = y.  MUST NOT be zero in the
 /// field, as f(0) = the shared secret.
 #[derive(Copy, Clone)]
-pub struct Identifier<C: Ciphersuite>(pub(crate) Scalar<C>);
+pub struct Identifier<C>(pub(crate) u16, PhantomData<C>);
 
-impl<C> AsRef<Scalar<C>> for Identifier<C>
+impl<C> Identifier<C>
 where
     C: Ciphersuite,
 {
-    fn as_ref(&self) -> &Scalar<C> {
-        &self.0
+    // Convert the identifier to a Scalar.
+    pub(crate) fn to_scalar(self) -> Scalar<C> {
+        // Classic left-to-right double-and-add algorithm that skips the first bit 1 (since
+        // identifiers are never zero, there is always a bit 1), thus `sum` starts with 1 too.
+        let one = <<C::Group as Group>::Field as Field>::one();
+        let mut sum = <<C::Group as Group>::Field as Field>::one();
+        let bits = (self.0.to_be_bytes().len() as u32) * 8;
+        for i in (0..(bits - self.0.leading_zeros() - 1)).rev() {
+            sum = sum + sum;
+            if self.0 & (1 << i) != 0 {
+                sum = sum + one;
+            }
+        }
+        sum
```

Comment:
I tried for hours to make this as a `From` impl but Rust doesn't let me.

(If you try to write a `From<Identifier<C>> for Scalar<C>`, Rust does not know that `Scalar<C>` can't be eventually `Identifier<C>` since it's a trait bound which could be possibly be satisfied by `Identifier<C>`.... I think? And it conflicts with the no-op blanket impl of `From<T> for T`)